### PR TITLE
In progress--Update travis.yml to run lint, followed by test.

### PR DESCRIPTION
**Pull request by phuchle**
Testing for travis config

### CODE REVIEW

**Comment by jwu910** [comment]
Doesn't travis run `npm test` by default? Or more so, wasn't travis already doing this?

This https://travis-ci.org/5-gwoap/adopt-a-family/jobs/419510910#L4001 build was from 3 days ago.

**Comment by phuchle** [comment]
I should have said initially that this PR isn't ready for review.  I'm submitting it so I can test that travis is doing what I want.

**Comment by jwu910** [comment]
@phuchle Gotcha travis is confusing lol

**Comment by rainrivas** [comment]
@phuchle FYI -- running `--fix` won't actually fix the source code since travis doesn't check code into the system. 
Also, do we want to force a line ending failure?

File: .editorconfig [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
-
-# Enforce line feed at end of line (instead of carriage return + line feed)
-# Prettier doesn't stricly enforce it so we do it here
-indent_style = space
-indent_size = 2
 end_of_line = lf
+indent_size = 2
+indent_style = space
+insert_final_newline = false
+max_line_length = 80
```
Comment (jwu910) [inline .editorconfig]: I think we can keep this property in the prettierrc. It's [not universally supported](https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties) on all editors' plugins.

Editorconfig's feature research blog state to just use [ruler guides](https://github.com/editorconfig/editorconfig/wiki/Property-research:-maximum-line-length) on your editor for this

Comment (jwu910) [reply .editorconfig]: Also, I think 80 is also prettier's default printWidth, my eslint used prettier rules to fix longer lines. We may not even need it

File: .travis.yml [hunk 2]
```diff
@@ -9,4 +9,7 @@ cache:
 services:
   - mongodb
 before_script:
-  - node server/seedDB.js
\ No newline at end of file
+  - node server/seedDB.js
+script:
+  - npm run test:lint
+  - npm run test
```
Comment (rainrivas) [inline .travis.yml]: Test calls lint so we don't need to have `run test:lint` in line 14 if we have this here.
test command: `"test": "concurrently 'npm run test:lint' 'npm run test:react' 'npm run test:api'",`

**Review by rainrivas — CHANGES_REQUESTED** [review]
Remove npm run test:lint or npm run test because it's repetetive

File: .travis.yml (hunk 2 above)
Comment (phuchle) [reply .travis.yml]: Good point.  What do you think about separating `test:lint` from `test`?  I feel like it's good to lint first explicitly so any lint errors will fail before having to run the tests.

Comment (jwu910) [reply .travis.yml]: We could have a separate test:all script to use concurrently. Doesn't the test:api need us to have our local db daemon up?

Comment (rainrivas) [reply .travis.yml]: Hmm I do think that having a separate lint call is important so lets pull the lint command out of test. Also, I do like the idea of creating a test all command, something meant to only test the src code logic -- `test:all`, `test:src`, `test:app`...
Thoughts @phuchle  @jwu910

Comment (phuchle) [reply .travis.yml]: Sounds gucci.  I'll separate the test logic up and have travis run each one by one.

**Comment by phuchle** [comment]
@rainrivas @jwu910 we gucci here?

**Review by rainrivas — APPROVED** [review]

**Review by jwu910 — APPROVED** [review]
Gucci

**Pull request merged**